keras_cv/layers/preprocessing/mix_up.py

```diff
@@ -79,6 +80,11 @@ def _batch_augment(self, inputs):
             )
             inputs["bounding_boxes"] = bounding_boxes
         inputs["images"] = images
+        if segmentation_masks is not None:
```

Comment:
can you perform input validation on the segmentation masks to ensure they are in one hot encoded format?  This can be done by inspecting the shape, and ensuring it matches [batch, h, w, num_classes].

Reply:
Yeah, I added input validation for segmentation_mask. Here is the [colab](https://colab.research.google.com/drive/1vz-5MbPgHta9l2FN6c9tgWu1WcyZLrHz?usp=sharing) showing the result.

Also, attaching the image below: 

<img width="1313" alt="Screenshot 2023-04-29 at 5 13 22 PM" src="https://user-images.githubusercontent.com/53268607/235300703-75e14ed3-7985-43c1-a196-850beb4b5d27.png">